google/cloud/spanner/ci/run_integration_tests_emulator_cmake.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+source "$(dirname "$0")/../../../../ci/lib/init.sh"
+source module lib/io.sh
+
+if [[ $# -lt 1 ]]; then
+  echo "Usage: $(basename "$0") <binary-dir> [ctest-args]"
+  exit 1
+fi
+
+# The first argument is where the build scripts have put the binaries, i.e., the
+# -B argument to the cmake configuration step.
+CMAKE_BINARY_DIR="$(
+  cd "${1}"
+  pwd
+)"
+readonly CMAKE_BINARY_DIR
+shift
+
+# Any additional arguments for the ctest invocation.
+ctest_args=("$@")
+
+spanner_emulator_pid=0
+if [[ -z "${CLOUD_SDK_LOCATION:-}" ]]; then
+  echo 1>&2 "You must set CLOUD_SDK_LOCATION to find the emulator"
+  exit 1
+fi
+
+# We cannot use `gcloud beta emulators spanner start` because there is no way
```

Comment:
This seems like a big misfeature of the `gcloud beta emulators`. Users should be able to write

```shell
gcloud beta emulators spanner start &
emulator_pid=$!
eval "$(gcloud beta emulators spanner env-init)"
# ... Run tests
kill $emulator_pid  # or maybe they need a `gcloud beta emulators spanner stop` command
```
And have it properly tear down the hierarchy of emulator processes. If that doesn't work, then the feature isn't very useful (as is shown by us not using it here). I'm happy to file a feature request for this if you agree that it would make things nicer.

Reply:
I agree, having `gcloud beta emulators spanner stop` would be a nice feature to have.